apps/web/pages/_app.tsx

```diff
@@ -17,18 +23,60 @@ export async function getInitialProps(context: AppContext) {
 }
 
 export default function StaticApp({ Component, pageProps }: AppProps) {
-  const router = useRouter();
+  // Cookie Manager Provider Configuration
+  const [isMounted, setIsMounted] = useState(false);
+  const trackingPreference = useRef<TrackingPreference | undefined>();
+
+  const setTrackingPreference = useCallback((newPreference: TrackingPreference) => {
```

Comment:
Worth considering extracting this to `libs/base-ui` package since it looks identical to the other usage, but not a blocker

Reply:
Thanks for calling that out. I hadn't noticed `libs/base-ui` yet. Should I think of that as a spot for components that we'll use across multiple apps in the monorepo?

In this case, are you suggesting I move the CookieManagerProvider component and all of it's props into libs/base-ui so I can just reuse it all as a component?

Reply:
I think U've got a good mental model of `base-ui`!

I'd actually only thought of extracting the `setTrackingPreference` as a hook, but now that U mention it, if U exported a component that wrapped that up along with all the other props that does seem even nicer.

Reply:
Cool, I'll give that a shot.

Do you know if there's an example of a base-ui component that's already being used in one of the apps? I'm trying to figure out the appropriate way to import a base-ui component/hook.

Reply:
We currently don't have any, however I believe U can import from the project like

`import { Button } from 'base-ui';`

Reply:
I gave that a shot, but I get the following error:
```
../../libs/base-ui/Button/index.tsx
Module parse failed: Unexpected token (8:7)
You may need an appropriate loader to handle this file type, currently no loaders are configured to process this file. See https://webpack.js.org/concepts#loaders
```

This happens whether I'm importing the Button or my own component. Maybe webpack needs to be configured to work with the ts/tsx files in `lib/base-ui`?

I'm not going to deploy this PR until Monday. If you understand how to adjust that config and want me to include it in this PR, let me know. Otherwise, I'll deploy this PR as-is on Monday and we can try to work out the base-ui bug as part of its own change.